salt/key.py

```diff
@@ -961,10 +961,12 @@ def _check_minions_directories(self):
         rejected = os.path.join(self.opts['pki_dir'], self.REJ)
         return accepted, pre, rejected, None
 
-    def check_minion_cache(self, preserve_minions=False):
+    def check_minion_cache(self, preserve_minions=None):
```

Comment:
If this parameter is not passed and a minion is not in `minions` then this will throw on line 987. The change to `None` matches the other implementation of this method.